05_vectors_and_result_enum.md

```diff
@@ -57,18 +73,37 @@ fn read_version(transaction_hex: &str) -> u32 {
 }
 ```
 
-What happens when we run `$ cargo run`? Well, we get an error. Take some time to read through it. You should see something like the following:
+What happens when we execute `$ cargo run`?
+
+Well, we get an error.
+Take some time to read through it.
+You should see something like the following:
+
 ```console
 error[E0608]: cannot index into a value of type `Result<Vec<u8>, FromHexError>`
 ```
 
-Let's examine what `hex::decode` returns. [Here is the doc](https://docs.rs/hex/latest/hex/fn.decode.html) for the `decode` method. Remember, we want to work with a Vector of bytes so a `vec<u8>` is the data type we're looking for. However, if we look at the return type of the `decode` function we see that the data structure we want is wrapped *inside* of a `Result` type. 
+Let's examine what `hex::decode` returns.
+[Here is the doc](https://docs.rs/hex/latest/hex/fn.decode.html) for the `decode` method.
+Remember, we want to work with a Vector of bytes so a `vec<u8>` is the data type we're looking for.
+However, if we look at the return type of the `decode` function we see that the data structure we want is wrapped *inside* of a `Result` type.
 
-The `Result` type is a common `enum` that you will see in Rust code. Enums are a way to describe a mutually exclusive set of options for a particular variable. If you think about it, it's possible that the `hex::decode` function fails to return a proper collection of bytes. For example, what if one of the characters is not a hex character? So we get two possibilities from a `Result`, an `Ok` response or an `Err` response.
+The `Result` type is a common `enum` that you will see in Rust code.
+Enums are a way to describe a mutually exclusive set of values for a particular variable.
+If you think about it, the `hex::decode` function can fail to return a proper collection of bytes.
+For example, what if one of the characters is not a hex character?
+So we get two possibilities from a `Result`, an `Ok` response or an `Err` response.
+The former represents a successful computation; the later represents that an error occurred.
```

Comment:
*latter

Maybe would change this to: "The former represents a successful computation whereas the latter indicates an error occurred"

Reply:
Surely better.